website/docs/tutorials/08-cairo-1-support/01-understanding-cairo-packages.md

```diff
@@ -0,0 +1,144 @@
+# Understanding Cairo Packages
+
+There are several requirements that Cairo packages have to follow. Further documentation pages assume some knowledge
+of Cairo package structure. Please refer to this page for details.
+
+## `.cairo` files
+
+Files with `.cairo` extension contain cairo code, including Starknet contracts. A file can define multiple methods,
+structures etc.
+
+## Modules
+
+Module consists of one or more cairo files, usually organized in a single directory. To define a module, create `.cairo`
+file named like the module and define components of this module with `mod` keyword.
+
+```
+my_module/
+├── a.cairo
+├── b.cairo
+└── c.cairo
+my_module.cairo
+```
+
+```cairo title="my_module.cairo"
+mod a;
+mod b;
+mod c;
+```
+
+Alternatively, modules can be defined within a file using
+
+```cairo title="my_module.cairo"
+mod my_module {
+    fn function_a() -> ();
+    // ...
+}
+```
+
+## Packages
+
+Package is a collection of modules that also defines `cairo_project.toml` and `lib.cairo` files.
+
+:::info
+Some other tools use name "crates" for packages.
+:::
+
+### `cairo_project.toml`
+
+It is needed for the definition of `[crate_roots]`, which is a directory containing `lib.cairo`.
+
+The default `cairo_project.toml` file contains only the definition of the `hello_starknet` package which is contained in
+the `src` directory.
+
+It is good practice for the package name defined here to match the top-level directory name.
+
+```toml title="cairo_project.toml"
+[crate_roots]
+hello_starknet = "src"
+```
+
+### `lib.cairo`
+
+It is the root of the package tree. Here you can define functions, declare used modules, etc.
+
+The default one has `contracts` and `business_logic` module declarations:
+
+```cairo title="lib.cairo"
+mod business_logic;
+mod contracts;
+```
+
+### Creating and using a new modules
+
+Suppose we wanted to create a module called `mod1` inside the `hello_starknet` package and use it in tests.
+We want this module to only have one file `functions.cairo` containing one function defined like:
+
+```cairo title="functions.cairo"
+fn returns_three() -> felt252 {
+    3
+}
+```
+
+Here is our initial file structure
+
+```
+my_project/
+└── hello_starknet/
+    ├── src/
+    │   ├── business_logic/
+    │   │   └── utils.cairo
+    │   ├── contracts/
+    │   │   └── hello_starknet.cairo
+    │   ├── business_logic.cairo
+    │   ├── contracts.cairo
+    │   └── lib.cairo
+    └── cairo_project.toml
```

Comment:
I would add two modules here though, to emphasise that packages is just modules packed together

Reply:
I'm not following, sorry. I think this can be simpler 🤔 I made a suggestion that would work and is far more simple than what you proposed. I understand that what you put here is probably consistent with a template you were doing in another pr but I still hold my opinion from the original comment - as a matter of fact I don't feel it got addressed in any of your comments unfortunately :( Could you go through my comment above and address both the suggestion and what I pointed out? Thanks :)

Reply:
I see. I still think this should be done as I said before.

> I think the structure you are suggesting is too simple for the purpose of this example

That's interesting, I think the very first example here should be as simple as possible. Here's the problem: I think the simplest example ever will be easier to understand for the users and you think that enforcing good practices from the very beginning of the docs is a good thing even if we start off with a little more complicated example.

I think we could meet halfway here. What about this: We could start with the simplest possible example and comment on it like "This is the minimum you need for this to work" and then we could say "But here's a better example with good practices and basically you should go with this."? I think that would come with the benefit of quality to this doc.

Reply:
On the second thought, I see your point with using minimal example here. We have an example better reflecting project structure on the Project initialization page.